src/objects/include/monetaryamount.hpp

```diff
@@ -189,6 +197,8 @@ class MonetaryAmount {
   int8_t _nbDecimals;
 };
 
+static_assert(sizeof(MonetaryAmount) == 16, "MonetaryAmount size should be optimized");
```

Comment:
<= would be more suited 
& what do you think of "MonetaryAmount size increase" as a assertion message ?